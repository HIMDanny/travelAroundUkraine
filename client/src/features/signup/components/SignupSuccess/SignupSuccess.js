/* eslint-disable react/prop-types */
import React from 'react';
import { Typography, Box } from '@mui/material';
import CheckCircleOutlineRoundedIcon from '@mui/icons-material/CheckCircleOutlineRounded';
import { useNavigate } from 'react-router-dom';
import { AlertModal } from '../../../../components';

const SignupSuccess = ({ open, onClose }) => {
  const navigate = useNavigate();

  const handleCloseBttn = () => {
    onClose();
    navigate('/catalogue');
  };

  const titleJsx = (
    <Box display="flex" justifyContent="center" alignItems="center" columnGap="5px">
      <Typography variant="h2" sx={{ color: 'success.main', mb: 0 }}>
        Sign up Successfull
      </Typography>
      <CheckCircleOutlineRoundedIcon fontSize="large" color="success" />
    </Box>
  );

  return (
    <AlertModal
      open={open}
      onClose={handleCloseBttn}
      onSubmit={handleCloseBttn}
      title={titleJsx}
      disableCancelButton
      submitButtonText="Great !"
      success
    >
      <Typography>You can enjoy your shopping with us now.</Typography>
    </AlertModal>
  );
};

export default SignupSuccess;
