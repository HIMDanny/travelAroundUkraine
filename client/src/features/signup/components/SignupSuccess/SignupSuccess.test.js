/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import SignupSuccess from './SignupSuccess';

jest.mock('@mui/material/Dialog', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogContent', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogActions', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/DialogTitle', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Button', () => ({ children }) => <button>{children}</button>);
jest.mock('@mui/icons-material/CheckCircleOutlineRounded', () => () => <div>SUCCESS ICON</div>);

const FormWrapper = ({ children }) => <MemoryRouter>{children} </MemoryRouter>;
const steps = ['User Info', 'Login Info'];

describe('SignupSuccess snapshot test', () => {
  test('should SignupSuccess snapshot render', () => {
    const { asFragment } = render(
      <FormWrapper>
        <SignupSuccess steps={steps} activeStep={steps.length} />
      </FormWrapper>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should SignupSuccess snapshot render null', () => {
    const { asFragment } = render(
      <FormWrapper>
        <SignupSuccess steps={steps} activeStep={0} />
      </FormWrapper>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
