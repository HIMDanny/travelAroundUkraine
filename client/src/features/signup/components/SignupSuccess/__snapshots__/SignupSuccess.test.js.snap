// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupSuccess snapshot test should SignupSuccess snapshot render 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <p>
          Sign up Successfull
        </p>
        <div>
          SUCCESS ICON
        </div>
      </div>
    </div>
    <div>
      <p />
      <p>
        You can enjoy your shopping with us now.
      </p>
      <p />
    </div>
    <div>
      <button>
        Great !
      </button>
    </div>
  </div>
   
</DocumentFragment>
`;

exports[`SignupSuccess snapshot test should SignupSuccess snapshot render null 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <p>
          Sign up Successfull
        </p>
        <div>
          SUCCESS ICON
        </div>
      </div>
    </div>
    <div>
      <p />
      <p>
        You can enjoy your shopping with us now.
      </p>
      <p />
    </div>
    <div>
      <button>
        Great !
      </button>
    </div>
  </div>
   
</DocumentFragment>
`;
